### ISSUE TITLE
The compiler doesn't warn about a non-nullable reference property in value types

### ISSUE THREAD

**Issue by YohDeadfall** [dotnet/roslyn#38580]
When nullable is enabled the compiler notifies that a reference type property in a class should be marked as nullable when there is no value assigned to it on initialization, but it doesn't do this for value types containing reference type properties. In the last case there is no way to specify the default value for a property since there is no parameterless constructor, and such properties are always initialized as `null`.

```csharp
#nullable enable

class ClassType
{
    public string Value { get; set; } // warning CS8618: Non-nullable property 'Value' is uninitialized. Consider declaring the property as nullable.
}

struct ValueType
{
    public string Value { get; set; } // no warning
}
```

Compiler version: '3.3.0-ci (<developer build>)'.
Language version: preview.

**Comment by jaredpar** [comment]
From the [NRT spec](https://github.com/dotnet/csharplang/blob/master/proposals/csharp-8.0/nullable-reference-types.md): 

> A warning is also given if a constructor does not explicitly initialize non-null reference fields.

For a `class` type there is an implicit empty constructor. The warning is generated because in this constructor the non-nullable property is not initialized. For a `struct` type there is no implicit empty constructor and one cannot be added. Hence no warnings are emitted for this case.

**Comment by YohDeadfall** [comment]
Isn't it dangerous? Structures as you know has no parameterless constructor. Therefore, initialization via `default` will set references to `null` which brakes safety introduced by nullable reference types.

**Comment by YohDeadfall** [comment]
By the way, makes sense to move the issue to the csharplang repo.
